components/dot_menu/dot_menu.jsx

```diff
@@ -116,124 +105,126 @@ export default class DotMenu extends Component {
+                            } else {
+                                this.props.actions.pinPost(this.props.post.id);
+                            }
                         }}
                     />
                 );
             }
         }
 
-        let dotMenuDelete = null;
-        if (this.state.canDelete) {
-            dotMenuDelete = (
+        if (canDelete) {
+            menuItems.push(
                 <DotMenuItem
-                    idPrefix={idPrefix + 'Delete'}
-                    isRHS={this.props.isRHS}
-                    idCount={this.props.idCount}
-                    post={this.props.post}
-                    commentCount={type === 'Post' ? this.props.commentCount : 0}
-                    actions={{
-                        openModal: this.props.actions.openModal,
+                    key={'delete'}
+                    menuItemText={<FormattedMessage id={'post_info.del'}/>}
+                    handleMenuItemActivated={(e) => {
+                        e.preventDefault();
+
+                        const deletePostModalData = {
+                            ModalId: ModalIdentifiers.DELETE_POST,
+                            dialogType: DeletePostModal,
+                            dialogProps: {
+                                post: this.props.post,
+                                commentCount: this.props.commentCount,
+                                isRHS: this.props.location === 'RHS_ROOT' || this.props.location === 'RHS_COMMENT',
+                            },
+                        };
+
+                        this.props.actions.openModal(deletePostModalData);
                     }}
                 />
             );
         }
 
-        let dotMenuEdit = null;
-        if (this.state.canEdit) {
-            dotMenuEdit = (
-                <DotMenuEdit
-                    idPrefix={idPrefix + 'Edit'}
-                    isRHS={this.props.isRHS}
-                    idCount={this.props.idCount}
-                    post={this.props.post}
-                    type={type}
-                    commentCount={type === 'Post' ? this.props.commentCount : 0}
-                    actions={{
-                        setEditingPost: this.props.actions.setEditingPost,
+        if (canEdit) {
+            menuItems.push(
+                <DotMenuItem
+                    key={'edit'}
+                    menuItemText={<FormattedMessage id={'post_info.edit'}/>}
+                    handleMenuItemActivated={() => {
+                        this.props.actions.setEditingPost(
+                            this.props.post.id,
+                            this.props.commentCount,
+                            this.props.location === 'CENTER' ? 'post_textbox' : 'reply_textbox',
+                            this.props.post.root_id && this.props.post.root_id.length > 0 ? Utils.localizeMessage('rhs_comment.comment', 'Comment') : Utils.localizeMessage('create_post.post', 'Post'),
```

Comment:
`this.props.post.root_id.length > 0` is redundant after checking `this.props.post.root_id` unless you just want that there for clarity. `this.props.post.root_id !== ''` would also work